packages/core/src/progress-circle/progress-circle.element.ts

```diff
@@ -115,21 +117,78 @@ export class CdsProgressCircle extends LitElement {
     }
   }
 
-  private setAriaAttributes() {
+  private _ariaLabel: string | undefined | null;
+
+  /**
+   * Updating the ariaLabelTemplate is an advanced edge case. Note that it will
+   * force a reset of the aria-label. If manipulating the aria label template,
+   * the assumption is that knowledge of this feature is deep enough to manage
+   * the side effects.
+   *
+   * @memberof CdsProgressCircle
+   */
+  @property({ type: String })
+  ariaLabelTemplate = '$i18n $val%';
+
+  /**
+   * The aria-label attribute is added here as a convenience. It is not set
+   * to a default value.
+   *
+   * If this attribute/property remains unset, it will default to an i18n string.
+   * This means that the aria-label can be customized using the aria-label
+   * attribute or by overriding the i18n value for the progress circle.
+   */
+  @property({ type: String })
+  get ariaLabel() {
+    return getAriaLabelOrDefault(
+      this._ariaLabel as string,
+      this.value,
+      this.ariaLabelTemplate,
+      this.i18n.loading,
+      this.i18n.looping
+    );
+  }
+
+  /**
+   * Changes the svg glyph displayed in the icon component. Defaults to the 'unknown' icon if
+   * the specified icon cannot be found in the icon registry.
+   */
+  set ariaLabel(newAriaLabel: string) {
+    if (hasStringPropertyChanged(newAriaLabel, this._ariaLabel as string)) {
+      const oldVal = this._ariaLabel;
+      this._ariaLabel = newAriaLabel;
+      this.requestUpdate('ariaLabel', oldVal);
+    }
+  }
+
+  @i18n() i18n = I18nService.keys.progress;
+
+  // note: this aria attr logic could be reused when we introduce progress bars
+  // consider promoting this to a shared utility in internal in the future
+  /** @private */
+  setAriaAttributes(oldValueForAriaLabelCheck?: number) {
     const currentValue = this.value;
-    const ariaAttrsIfValue: HTMLAttributeTuple[] = [
-      ['role', 'progressbar'],
-      ['aria-valuemin', '0'],
-      ['aria-valuemax', '100'],
-      ['aria-valuenow', currentValue + ''],
-    ];
-    const ariaAttrsNoValue: HTMLAttributeTuple[] = [
-      ['role', 'img'],
-      ['aria-valuemin', false],
-      ['aria-valuemax', false],
-      ['aria-valuenow', false],
-    ];
-    const attrsToSet = isNil(currentValue) ? ariaAttrsNoValue : ariaAttrsIfValue;
+    const oldValIsNil = isNilOrEmpty(oldValueForAriaLabelCheck);
+    let ariaLabel: string;
+
+    if (oldValIsNil && !isNilOrEmpty(currentValue) && this.ariaLabel === this.i18n.looping) {
+      if (!isNilOrEmpty(currentValue) && this.ariaLabel === this.i18n.looping) {
```

Comment:
Isn't this "if" already included in the above? Is it possible to get in the else case at all?
Maybe we can extract this same line to a named method, helping for readability. At least that's the advise I got from you for a similar block of code in a previous PR :)

Reply:
good catch. i simplified this logic. ❤️